packages/ui5-application-writer/src/data/defaults.ts

```diff
@@ -113,8 +140,13 @@ function getLocalVersion({
             result =
                 framework === 'SAPUI5' ? UI5_DEFAULT.MIN_LOCAL_SAPUI5_VERSION : UI5_DEFAULT.MIN_LOCAL_OPENUI5_VERSION; // minimum version available as local libs
         }
-        if (parseFloat(version) > parseFloat(result)) {
-            result = version;
+
+        if (version !== UI5_DEFAULT.DEFAULT_UI5_VERSION) {
+
+            // Update to a valid coerced version string e.g. 1.80 -> 1.80.0. Cannot be null as previously validated.
+            if (semVer.gt(semVer.coerce(version)!, semVer.coerce(result)!)) {
+                result = version;
+            }
```

Comment:
Is the `else` before the addition (line 140) the same as the this condition: `if (version !== UI5_DEFAULT.DEFAULT_UI5_VERSION)`? If so, there seems to be a chance an open UI5 version could get overwritten by the addition.

Reply:
I believe `result` would be overwritten by the addition, at least that's how it was previously implemented. But I see we can refactor here to make the intention clearer.